src/versions/drafts/buildDraftMergeAggregate.ts

```diff
@@ -0,0 +1,79 @@
+import { PipelineStage } from 'mongoose';
+import { SanitizedCollectionConfig } from '../../collections/config/types';
+
+type Args = {
+  config: SanitizedCollectionConfig
+  query: Record<string, unknown>
+}
+
+export const buildDraftMergeAggregate = ({ config, query }: Args): PipelineStage[] => [
+  // Merge in versions by parent ID
+  {
+    $lookup: {
+      from: `_${config.slug}_versions`,
+      as: '_versions',
+      localField: '_id',
+      foreignField: 'parent',
+    },
+  },
+
+  // Add a new field
+  // with the last (newest) version returned
+  {
+    $addFields: {
+      _version: {
+        $arrayElemAt: ['$_versions', -1],
+      },
+    },
+  },
+
+  // If newer version exists,
+  // merge the version into the root and
+  // replace the updatedAt
+  // Otherwise, do nothing to the root
+  {
+    $replaceRoot: {
+      newRoot: {
+        $cond: {
+          if: {
+            $and: [
+              {
+                $ne: ['$_version', undefined],
```

Comment:
does this work if there are no versions?

$arrayElemAt can be either null or not defined - i am not sure if not defined is the same as undefined

[See docs](https://www.mongodb.com/docs/manual/reference/operator/aggregation/arrayElemAt/#behavior)